### ISSUE TITLE
Allow default parameters in no-empty-function

### ISSUE THREAD

**Issue by steelbrain** [eslint/eslint#12718]
**What rule do you want to change?**
`no-empty-function`

**Does this change cause the rule to produce more or fewer warnings?**
It causes the rule to produce fewer warnings, **when configured to do so**, default behavior stays the same

**How will the change be implemented? (New option, new default behavior, etc.)?**

The `allow` option be amended to allow the value `defaultValue` or `defaultParameter` (or option, you get the idea), when that rule is allowed the lint rule should ignore all empty functions that are defined as parameter defaults.

**Please provide some example code that this change will affect:**

```js
// Warn users (also, current behavior):
// "allow": []
function something(a = () => {}) {
  console.log('Hello World')
}
function another({
  parameter = () => {}
}) {
  console.log('Hello World')
}

// Good (because option is specified)
// "allow": ["defaultValue"]
function something(a = () => {}) {
  console.log('Hello World')
}
function another({
  parameter = () => {}
}) {
  console.log('Hello World')
}
```

**What does the rule currently do for this code?**
As mentioned in code above, rule always complains

**What will the rule do after it's changed?**
If the option is specified, not complain for default value scenarios

**Are you willing to submit a pull request to implement this change?**

Yes, I am, unless somebody beats me to it

**Comment by kaicataldo** [comment]
I think it would be safer to explicitly denote no-ops using an inline comment instead. It would also allow users to denote no-ops in other situations as well.

```js
function fn(/* no-op */ a = () => {}) {
  console.log('Hello World')
}

function fn2(a) {
 return a || /* no-op */ function() {};
}

a('b', /* no-op */ () => {});
```

**Comment by kaicataldo** [comment]
That being said, I'm not sure it's that much less effort than just disabling the rule for that line.

**Comment by steelbrain** [comment]
While it's true that we can add a no-op inside every default function body, or eslint ignore lines on all such instances. Wouldn't it still be nice to not have to do that and be able to configure that in the global config? Especially considering that it's a trivial change in the plugin sources.

I'm willing to submit a PR if you like the idea

**Comment by kaicataldo** [comment]
My concern is that it only solves the problem in one place (default parameter values), and I don't think that's a case that is any more valid than the other examples I've listed above. I'm not against this, but I do think it makes sense to have a more generalized solution that will allow for the other cases as well.__
